app/services/metadata/handlers/teacher.rb

```diff
@@ -68,5 +71,9 @@ def latest_mentor_training_period_by_lead_provider(teacher:)
       .order("lead_provider_id, training_periods.started_on DESC")
       .index_by { it.lead_provider&.id }
     end
+
+    def school_transfers_for(school_periods, lead_provider_id)
+      ::Teachers::SchoolTransfers::History.transfers_for(school_periods:, lead_provider_id:)
```

Comment:
I don't think this will be a bottleneck, but if it is we could maybe add a `involved_in_school_tranfer?` method to `History` that is more optimal/returns after finding the first transfer for the LP.